https://github.com/influxdata/telegraf/pull/17626

From de4d0c23121e476882b368d921742f3ae0506a4f Mon Sep 17 00:00:00 2001
From: Konstantin <k.kulikov2@gmail.com>
Date: Thu, 11 Sep 2025 17:50:58 +0300
Subject: [PATCH] fix(inputs.whois): fix flaky test

Running whois tests with "go test -count 10000 ." results in panic below. I don't see any reason to set listener to nil.

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc85580]

goroutine 66840 [running]:
github.com/influxdata/telegraf/plugins/inputs/whois.(*server).start.func1()
	/home/kaey/.cache/aport-bump/telegraf/plugins/inputs/whois/whois_test.go:190 +0x40
created by github.com/influxdata/telegraf/plugins/inputs/whois.(*server).start in goroutine 66884
	/home/kaey/.cache/aport-bump/telegraf/plugins/inputs/whois/whois_test.go:188 +0xfc
FAIL	github.com/influxdata/telegraf/plugins/inputs/whois	2.669s
FAIL
---
 plugins/inputs/whois/whois_test.go | 1 -
 1 file changed, 1 deletion(-)

diff --git a/plugins/inputs/whois/whois_test.go b/plugins/inputs/whois/whois_test.go
index 284e9660b8cbf..63007e689f20e 100644
--- a/plugins/inputs/whois/whois_test.go
+++ b/plugins/inputs/whois/whois_test.go
@@ -222,6 +222,5 @@ func (s *server) start() (string, error) {
 func (s *server) stop() {
 	if s.listener != nil {
 		s.listener.Close()
-		s.listener = nil
 	}
 }
